src/main/java/org/isf/medicalinventory/gui/InventoryWardBrowser.java

```diff
@@ -0,0 +1,359 @@
+                    jTableInventory.setModel(new InventoryBrowsingModel());
+                }
+            });
+        }
+        return jCalendarFrom;
+    }
+
+    private JDateChooser getJCalendarTo() {
+        if (jCalendarTo == null) {
+            dateTo.set(GregorianCalendar.HOUR_OF_DAY, 23);
+            dateTo.set(GregorianCalendar.MINUTE, 59);
+            dateTo.set(GregorianCalendar.SECOND, 59);
+            jCalendarTo = new JDateChooser(dateTo.getTime()); // Calendar
+            jCalendarTo.setDateFormatString("dd/MM/yy"); //$NON-NLS-1$
+            jCalendarTo.addPropertyChangeListener("date", new PropertyChangeListener() { //$NON-NLS-1$
+
+                public void propertyChange(PropertyChangeEvent evt) {
+                    jCalendarTo.setDate((Date) evt.getNewValue());
+                    dateTo.setTime((Date) evt.getNewValue());
+                    dateTo.set(GregorianCalendar.HOUR_OF_DAY, 23);
+                    dateTo.set(GregorianCalendar.MINUTE, 59);
+                    dateTo.set(GregorianCalendar.SECOND, 59);
+                    jTableInventory.setModel(new InventoryBrowsingModel());
+                }
+            });
+        }
+        return jCalendarTo;
+    }
+
+    private JLabel getJLabelTo() {
+        if (jLabelTo == null) {
+            jLabelTo = new JLabel();
+            jLabelTo.setHorizontalAlignment(SwingConstants.RIGHT);
+            jLabelTo.setText(MessageBundle.getMessage("angal.billbrowser.to")); //$NON-NLS-1$
+        }
+        return jLabelTo;
+    }
+
+    private JLabel getJLabelFrom() {
+        if (jLabelFrom == null) {
+            jLabelFrom = new JLabel();
+            jLabelFrom.setHorizontalAlignment(SwingConstants.RIGHT);
+            jLabelFrom.setText(MessageBundle.getMessage("angal.billbrowser.from")); //$NON-NLS-1$
+        }
+        return jLabelFrom;
+    }
+
+    private JButton getNewButton() {
+        newButton = new JButton(MessageBundle.getMessage("angal.inventory.new"));
+        newButton.setMnemonic(KeyEvent.VK_N);
+        return newButton;
+    }
+
+    private JButton getViewButton() {
+        viewButton = new JButton(MessageBundle.getMessage("angal.inventory.view"));
+        viewButton.setMnemonic(KeyEvent.VK_V);
+        return viewButton;
+    }
+
+    private JButton getUpdateButton() {
+        updateButton = new JButton(MessageBundle.getMessage("angal.inventory.update"));
+        updateButton.setMnemonic(KeyEvent.VK_M);
+        return updateButton;
+    }
+
+    private JButton getDeleteButton() {
+        deleteButton = new JButton(MessageBundle.getMessage("angal.inventory.delete"));
+        deleteButton.setMnemonic(KeyEvent.VK_D);
+        return deleteButton;
+    }
+
+    private JButton getCloseButton() {
+        closeButton = new JButton(MessageBundle.getMessage("angal.inventory.close"));
+        closeButton.setMnemonic(KeyEvent.VK_C);
+        closeButton.addActionListener(new ActionListener() {
```

Comment:
Only the **Close** button seems to have some action associated with it; that doesn't feel correct.

Reply:
I think the whole PR is not meant to land on develop but on a "feature-branch" as I've suggested on Slack.
Since this work seems to be a porting of a Cameroon customization, this division in chunks could be useful to verify and merge the whole work in small pieces at time though, anyway I'm suggesting (and waiting) more organization, like to have a parent issues for all these chunks.

In few words: next button actions will come with future PRs